owncloudApp/src/main/java/com/owncloud/android/ui/fragment/FileDetailFragment.java

```diff
@@ -329,11 +332,33 @@ public void onClick(View v) {
                 ((FileDisplayActivity) mContainerActivity).cancelTransference(getFile());
                 break;
             }
+            case R.id.fdIcon: {
+                displayFile(getFile());
+            }
             default:
                 Timber.e("Incorrect view clicked!");
         }
     }
 
+    private void displayFile(OCFile file) {
+        if (PreviewImageFragment.canBePreviewed(file)) {
+            // preview image - it handles the sync, if needed
+            ((FileDisplayActivity) mContainerActivity).startImagePreview(file);
+        } else if (PreviewTextFragment.canBePreviewed(file)) {
+            ((FileDisplayActivity) mContainerActivity).startTextPreview(file);
+            mContainerActivity.getFileOperationsHelper().syncFile(file);
+
+        } else if (PreviewAudioFragment.canBePreviewed(file)) {
+            // media preview
+            ((FileDisplayActivity) mContainerActivity).startAudioPreview(file, 0);
+            mContainerActivity.getFileOperationsHelper().syncFile(file);
+
+        } else {
+            // sync file content, then open with external apps
+            ((FileDisplayActivity) mContainerActivity).startSyncThenOpen(file);
+        }
+    }
```

Comment:
I would say I have seen this logic before. I'll refactor this within the new arch 🏗️

Reply:
Yeah, it is quite similar, but in this case, the video check is not necessary, be careful in the future 🤞🏻